src/cache/cache_las.c

```diff
@@ -1032,93 +1002,62 @@ __wt_find_lookaside_upd(
         WT_ERR(las_cursor->get_value(
           las_cursor, &durable_timestamp, &prepare_state, &upd_type, las_value));
 
+        /* We do not have have prepared updates in the lookaside anymore */
+        WT_ASSERT(session, prepare_state != WT_PREPARE_INPROGRESS);
+
         /*
          * Found a visible record, return success unless it is prepared and we are not ignoring
          * prepared.
          *
          * It's necessary to explicitly signal a prepare conflict so that the callers don't fallback
          * to using something from the update list.
+         *
+         * FIXME-PM-1521: To be removed or reverted in the future
          */
-        if (prepare_state == WT_PREPARE_INPROGRESS &&
-          !F_ISSET(&session->txn, WT_TXN_IGNORE_PREPARE) && !allow_prepare) {
-            ret = WT_PREPARE_CONFLICT;
-            break;
-        }
+        // if (prepare_state == WT_PREPARE_INPROGRESS &&
+        //   !F_ISSET(&session->txn, WT_TXN_IGNORE_PREPARE) && !allow_prepare) {
+        //     ret = WT_PREPARE_CONFLICT;
+        //     break;
+        // }
 
-        /* We do not have birthmarks in the lookaside anymore. */
-        WT_ASSERT(session, upd_type != WT_UPDATE_BIRTHMARK);
+        /* We do not have birthmarks and tombstones in the lookaside anymore. */
+        WT_ASSERT(session, upd_type != WT_UPDATE_BIRTHMARK && upd_type != WT_UPDATE_TOMBSTONE);
 
         /*
          * Keep walking until we get a non-modify update. Once we get to that point, squash the
          * updates together.
          */
         if (upd_type == WT_UPDATE_MODIFY) {
             WT_NOT_READ(modify, true);
-            __find_birthmark_update(cbt, &birthmark_upd);
             while (upd_type == WT_UPDATE_MODIFY) {
                 WT_ERR(__wt_update_alloc(session, las_value, &mod_upd, &notused, upd_type));
                 WT_ERR(__wt_modify_vector_push(&modifies, mod_upd));
                 mod_upd = NULL;
 
                 /*
-                 * This loop can exit on three conditions:
-                 * 1: We find a standard update in the lookaside table to apply the deltas to.
-                 * 2: We find that the birthmark record for the given key is more recent than the
-                 * current lookaside record.
-                 * 3: We cross the key boundary meaning that the base update MUST be the on-disk
-                 * value for that key. Note that if we hit the beginning of the lookaside table when
-                 * backtracking, this equates to crossing a key boundary.
+                 * Find the base update to apply the reverse deltas
                  */
-                WT_ERR_NOTFOUND_OK(las_cursor->prev(las_cursor));
+                WT_ERR(las_cursor->next(las_cursor));
```

Comment:
It is possible that there can be no record in the `las`. Why is that an error scenario?

Reply:
Good catch, I think I changed it by accident.